ambry-network/src/main/java/com.github.ambry.network/BlockingChannel.java

```diff
@@ -102,6 +102,20 @@ public void disconnect() {
     }
   }
 
+  public void reset() {
```

Comment:
After revisiting the code, I think you need to make changes in both `BlockingChannel` and `SSLBlockingChannel`.

Reply:
Never mind, I see `SSLBlockingChannel` extends `BlockingChannel`

Reply:
The reset() is inherited by SSLBlockingChannel, and all reset() is doing is setting setSoLinger and then calling the disxonnet() method of the object, which, in case of SSLBlockingChannel would be its own disconnect().